lib/trino-parquet/src/main/java/io/trino/parquet/reader/SimpleSliceInputStream.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader;
+
+import io.airlift.slice.Slice;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Basic input stream based on a given Slice object.
+ * This is a simpler version of BasicSliceInput with a few additional methods.
```

Comment:
What additional methods? Why do we need this new abstraction at all?

Reply:
Right now it's only 2 methods getByteArray and getByteArrayOffset.
In subsequent PRs when we start adding our own implementations of values decoders, we'll be adding usage of `UnsafeSlice` methods as those were found to perform better in benchmarks.
@skrzypo987 might know more details.